mozci/push.py

```diff
@@ -70,10 +74,35 @@ class Push:
 
     def __init__(self, revs, branch="autoland"):
         if isinstance(revs, str):
+            # Direct usage of a single revision reference
             self._revs = None
             revs = [revs]
+            self._bugs = None
         else:
-            self._revs = revs
+
+            # We may get detailed revision objects here
+            if all(map(lambda r: isinstance(r, dict), revs)):
+
+                def _parse_bug(rev):
```

Comment:
You can use https://hg.mozilla.org/hgcustom/version-control-tools/file/9822fcf4b1178d219b7d7a386dda02a11facf55b/pylib/mozautomation/mozautomation/commitparser.py#l97 to have exactly the same behavior as before.

Unrelated question: can you move the function to the top-level?

Reply:
Done